docs/upgrading.md

```diff
@@ -1,66 +1,104 @@
+If you are running using the `ds-hostnet-split` example or a derivative, we strongly recommend
+that you generate new certificates for securing your gRPC communication between Contour and Envoy.
+
+### Other changes
+
+The other major feature that is available is an experimental version of Leader Election.
+Currently, you can enable leader election with the `--enable-leader-election` flag to
+`contour serve`. This will ensure that *only the elected leader serves the gRPC endpoints.*
+That is, only the elected leader will actually open the gRPC port and start serving.
+
+If you have deployed Contour and Envoy in their own pods - we call this split deployment - you
+should enable leader election so all envoy pods take their configuration from the lead contour. 
 
-If you are running using the `ds-hostnet-split` example or a derivative, we strongly recommend that you generate new certificates for securing your gRPC communication between Contour and Envoy.
+There are some further requirements for this, see the [howto](#enabling-leader-election) section below for more.
 
-There is a Job in the `ds-hostnet-split` directory that will use the new `contour certgen` command to generate a CA and then sign Contour and Envoy keypairs, which can also then be saved directly to Kubernetes as Secrets, ready to be mounted into your Contour and Envoy Deployments and Daemonsets.
+### Enabling TLS for gRPC
+
+There is a Job in the `ds-hostnet-split` directory that will use the new `contour certgen` command
+to generate a CA and then sign Contour and Envoy keypairs, which can also then be saved directly
+to Kubernetes as Secrets, ready to be mounted into your Contour and Envoy Deployments and Daemonsets.
 
 Again, check the examples for Contour and Envoy in `ds-hostnet-split` for the exact commands.
 
-The steps you will need to follow for this to work are detailed more in (grpc-tls-howto.md)[./grpc-tls-howto.md]:
+If you wish to do this manually, you can use the steps detailed in (grpc-tls-howto.md)[./grpc-tls-howto.md].
+
+In short, you will need to:
 1. Generate a CA keypair
-1. Use it to generate and sign two more keypairs, one for Contour and one for Envoy.
-1. Get the CA certificate (only) and the Contour and Envoy keypairs into Kubernetes Secrets inside the `heptio-contour` (or whatever) namespace.
-1. Mount the CA cert and the keypair into the Contour and Envoy pods, and supply the correct parameters to `contour serve` (for Contour) and `contour bootstrap` (for Envoy)
+2. Use it to generate and sign two more keypairs, one for Contour and one for Envoy.
+3. Get the CA certificate (only) and the Contour and Envoy keypairs into Kubernetes Secrets inside the `heptio-contour` (or whatever) namespace.
+5. Mount the CA cert and the keypair into the Contour and Envoy pods, and supply the correct parameters to `contour serve` (for Contour) and `contour bootstrap` (for Envoy)
+
+As mentioned in (grpc-tls-howto.md)[./grpc-tls-howto.md], the `contour certgen` command can
```

Comment:
It's more important that this document is accurate than comprehensive. Don't duplicate information as that could lead to people having more or less information depending on the page they read. Perhaps the right thing to do is to move the additions to ./grpc-tls-howto.md to this page and make them canonical.